htdocs/core/db/mysqli.class.php

```diff
@@ -678,7 +678,7 @@ public function DDLListTables($database, $table = '')
 		}
 		$tmpdatabase = preg_replace('/[^a-z0-9\.\-\_]/i', '', $database);
 
-		$sql = "SHOW TABLES FROM ".$tmpdatabase." ".$like.";";
+		$sql = "SHOW TABLES FROM `".$tmpdatabase."` ".$like.";";
```

Comment:
On line 679, we clean database name to remove - char 
So i am surprised your fix work. If thee is a - into name (we try to avoid this in install process), the show tables will use table name without the -

Reply:
The regex is removing anything that's not a to z, 0 to 9, period, hyphen or underscore (case insensitive).
Also, I don't see where in the install process the - or other bad characters are stripped or checked.

The GETPOST filter aZ09 also checks with the same regex.

```php
$db_name = GETPOST('db_name', 'aZ09') ?GETPOST('db_name', 'aZ09') : (empty($argv[10]) ? '' : $argv[10]);
```

```php
case 'aZ09':
  if (!is_array($out)) {
    $out = trim($out);
      if (preg_match('/[^a-z0-9_\-\.]+/i', $out)) {
          $out = '';
      }
  }
  break;
```

Also, in `DDLCreateDb()` we do use backticks:

```php
public function DDLCreateDb($database, $charset = '', $collation = '', $owner = '')
{
	// phpcs:enable
	if (empty($charset)) {
		$charset = $this->forcecharset;
	}
	if (empty($collation)) {
		$collation = $this->forcecollate;
	}

	// ALTER DATABASE dolibarr_db DEFAULT CHARACTER SET latin DEFAULT COLLATE latin1_swedish_ci
	$sql = "CREATE DATABASE `".$this->escape($database)."`";
	$sql .= " DEFAULT CHARACTER SET `".$this->escape($charset)."` DEFAULT COLLATE `".$this->escape($collation)."`";

	dol_syslog($sql, LOG_DEBUG);
	$ret = $this->query($sql);
	if (!$ret) {
		// We try again for compatibility with Mysql < 4.1.1
		$sql = "CREATE DATABASE `".$this->escape($database)."`";
		dol_syslog($sql, LOG_DEBUG);
		$ret = $this->query($sql);
	}
	return $ret;
}
```

Reply:
![image](https://user-images.githubusercontent.com/7734808/214778816-ecca6b4e-e4e3-4b12-8c5a-8006616d611d.png)